src/EventSubscriber/GenericEventSubscriber.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\EventSubscriber\GenericEventSubscriber.
+ */
+
+namespace Drupal\rules\EventSubscriber;
+
+use Drupal\Core\Entity\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * Subsscribes to Symfony events and maps them to Rules events.
+ */
+class GenericEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The entity manager used for loading reaction rule config entities.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager) {
+    $this->entityManager = $entity_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    // Register this listener for every event that is used by a reaction rule.
+    $events = [];
+    $callback = ['onRulesEvent', 100];
+
+    // If there is no state service there is nothing we can do here.
+    if (!\Drupal::hasService('state')) {
+      return [];
+    }
+
+    // Since we cannot access the reaction rule config storage here we have to
```

Comment:
Will add the comment "This static method could be called early when the container is built, so the state service might no always be available."